sky/packages/sky/lib/src/widgets/framework.dart

```diff
@@ -824,29 +825,36 @@ abstract class Component extends Widget {
     String _debugPreviousComponent;
     assert(() {
       _debugIsBuilding = true;
-      _debugPreviousComponent = _debugLastComponent;
-      if (_debugLastComponent != null)
-        _debugLastComponent = "$_debugPreviousComponent -> ${this.toStringName()}";
-      else
-        _debugLastComponent = "Build chain: ${this.toStringName()}";
+      _debugComponentBuildTree.add(this);
       return true;
     });
 
     int lastOrder = _currentOrder;
     _currentOrder = _order;
-    _child = build();
+    try {
+      _child = build();
+      assert(_child != null);
+    } catch (e, stack) {
+      debugReportException("building ${this.toStringName()}", e, stack);
+    }
     _currentOrder = lastOrder;
-    assert(_child != null);
     assert(() { _debugChildTaken = false; return true; });
-    _child = syncChild(_child, oldChild, slot);
-    assert(!_debugChildTaken); // we shouldn't be able to lose our child when we're syncing it!
-    assert(_child != null);
-    assert(_child.parent == this);
-
+    try {
+      _child = syncChild(_child, oldChild, slot);
```

Comment:
We shouldn't bother to call syncChild if _child is null.  We're just going to get garbage.